src/auto/dbus_message.rs

```diff
@@ -0,0 +1,427 @@
+    }
+
+    pub fn get_byte_order(&self) -> DBusMessageByteOrder {
+        unsafe {
+            from_glib(gio_sys::g_dbus_message_get_byte_order(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_destination(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_message_get_destination(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_error_name(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_message_get_error_name(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn get_flags(&self) -> DBusMessageFlags {
+        unsafe { from_glib(gio_sys::g_dbus_message_get_flags(self.to_glib_none().0)) }
+    }
+
+    pub fn get_header(&self, header_field: DBusMessageHeaderField) -> Option<glib::Variant> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_message_get_header(
+                self.to_glib_none().0,
+                header_field.to_glib(),
+            ))
+        }
+    }
+
+    pub fn get_interface(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_message_get_interface(self.to_glib_none().0)) }
+    }
+
+    pub fn get_locked(&self) -> bool {
+        unsafe { from_glib(gio_sys::g_dbus_message_get_locked(self.to_glib_none().0)) }
+    }
+
+    pub fn get_member(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_message_get_member(self.to_glib_none().0)) }
+    }
+
+    pub fn get_message_type(&self) -> DBusMessageType {
+        unsafe {
+            from_glib(gio_sys::g_dbus_message_get_message_type(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    #[cfg(any(unix, feature = "dox"))]
+    pub fn get_num_unix_fds(&self) -> u32 {
+        unsafe { gio_sys::g_dbus_message_get_num_unix_fds(self.to_glib_none().0) }
+    }
+
+    pub fn get_path(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_message_get_path(self.to_glib_none().0)) }
+    }
+
+    pub fn get_reply_serial(&self) -> u32 {
+        unsafe { gio_sys::g_dbus_message_get_reply_serial(self.to_glib_none().0) }
+    }
+
+    pub fn get_sender(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_message_get_sender(self.to_glib_none().0)) }
+    }
+
+    pub fn get_serial(&self) -> u32 {
+        unsafe { gio_sys::g_dbus_message_get_serial(self.to_glib_none().0) }
+    }
+
+    pub fn get_signature(&self) -> Option<GString> {
+        unsafe { from_glib_none(gio_sys::g_dbus_message_get_signature(self.to_glib_none().0)) }
+    }
+
+    #[cfg(any(unix, feature = "dox"))]
+    pub fn get_unix_fd_list(&self) -> Option<UnixFDList> {
+        unsafe {
+            from_glib_none(gio_sys::g_dbus_message_get_unix_fd_list(
+                self.to_glib_none().0,
+            ))
+        }
+    }
+
+    pub fn lock(&self) {
```

Comment:
Question: what is "lock" used for? If some operations cannot be done if the message is unlocked, then this method should return a wrapper which implements those methods (actually, maybe it should return a wrapper anyway).

Reply:
Apparently locking just makes mutations into no-ops that warn:

https://gitlab.gnome.org/GNOME/glib/-/blob/c93318953eec6a65de3e466f648a4f26109ff0eb/gio/gdbusmessage.c#L835-840

Reply:
This should probably be manually implemented similar to `MutexGuard`: have some object returned that on `Drop` unlocks

Reply:
There is no `unlock`!

This is like a `freeze` thing that makes the actual object behind the ref count immutable and mutations become no-ops.

Reply:
This whole thing makes no sense... :rofl:

Reply:
Ah I see. In that case it should be fine like this. Ugly that it just prints a warning and doesn't return an actual error otherwise but changing that on our side would require to implement all these manually and first check whether the message is locked.

Reply:
> This whole thing makes no sense...

It makes a lot of sense, just that this being C the API is awful. In Rust you would have a builder (or otherwise two different types), and then at some point "build" the message and get an actual (locked, frozen, immutable) message. Here you can modify things and then at some point decide to lock it to make it immutable. Ugly but that's how it is.